rust/processor/src/processors/parquet_processors/parquet_transaction_metadata_processor.rs

```diff
@@ -0,0 +1,147 @@
+            config.bucket_name.clone(),
+            config.bucket_root.clone(),
+            config.parquet_handler_response_channel_size,
+            config.max_buffer_size,
+            config.parquet_upload_interval_in_secs(),
+        );
+        Self {
+            connection_pool,
+            write_set_size_info_sender,
+        }
+    }
+}
+
+impl Debug for ParquetTransactionMetadataProcessor {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "ParquetTransactionMetadataProcessor {{ capacity of write set size info channel: {:?} }}",
+            self.write_set_size_info_sender.capacity(),
+        )
+    }
+}
+
+#[async_trait]
+impl ProcessorTrait for ParquetTransactionMetadataProcessor {
+    fn name(&self) -> &'static str {
+        ProcessorName::ParquetTransactionMetadataProcessor.into()
+    }
+
+    async fn process_transactions(
+        &self,
+        transactions: Vec<Transaction>,
+        start_version: u64,
+        end_version: u64,
+        _: Option<u64>,
+    ) -> anyhow::Result<ProcessingResult> {
+        let last_transaction_timestamp = transactions.last().unwrap().timestamp.clone();
+        let mut transaction_version_to_struct_count: AHashMap<i64, i64> = AHashMap::new();
+
+        let mut write_set_sizes = vec![];
+
+        for txn in &transactions {
+            let txn_version = txn.version as i64;
+            let size_info = match txn.size_info.as_ref() {
+                Some(size_info) => size_info,
+                None => {
+                    warn!(version = txn.version, "Transaction size info not found");
+                    continue;
+                },
+            };
+            for (index, write_set_size_info) in size_info.write_op_size_info.iter().enumerate() {
+                write_set_sizes.push(WriteSetSize::from_transaction_info(
+                    write_set_size_info,
+                    txn_version,
+                    index as i64,
+                ));
+                transaction_version_to_struct_count
+                    .entry(txn_version)
+                    .and_modify(|e| *e += 1)
+                    .or_insert(1);
+            }
+        }
+
+        let write_set_size_info_parquet_data = ParquetDataGeneric {
+            data: write_set_sizes,
+            transaction_version_to_struct_count: transaction_version_to_struct_count.clone(),
+        };
+
+        self.write_set_size_info_sender
+            .send(write_set_size_info_parquet_data)
+            .await
+            .context("Error sending write set size info to parquet handler")?;
+
+        Ok(ProcessingResult::ParquetProcessingResult(
+            ParquetProcessingResult {
+                start_version: start_version as i64,
+                end_version: end_version as i64,
+                last_transaction_timestamp: last_transaction_timestamp.clone(),
+                txn_version_to_struct_count: AHashMap::new(),
```

Comment:
Do we need to give it `transaction_version_to_struct_count`?

Reply:
here we don't have to. I think this is a flaw that we would like to improve. b/c transaction_version_to_struct_count is only being used when the structs are processed (insert into buffer) in the parquet handler.